### PR TITLE
feat: react based studio footer

### PR DESCRIPTION
**Pull request by KristinAoki**
This a rebuild of the current studio footer using React. The footer should have the same content as the existing footer.

For testing use course-authoring [PR 532](https://github.com/openedx/frontend-app-course-authoring/pull/532). The footer can be seen on at the bottom of Pages & Resources.

### CODE REVIEW

File: src/footer/Footer.jsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+  platformName,
+  lmsBaseUrl,
+  studioBaseUrl,
+  showAccessibilityPage,
+  // injected
+  intl,
+}) => {
+  const [isOpen, setIsOpen] = useState(false);
+
+  return (
```
Comment (connorhaugh) [inline src/footer/Footer.jsx:29]: I'm assuming the giant pile of jsx is not a choice you would make, but this is copied from the header package?

File: src/footer/Footer.test.jsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+      expect(screen.getByText(messages.closeHelpButtonLabel.defaultMessage))
+        .toBeVisible();
+    });
+    it('help button link row should be visible', () => {
+      renderComponent();
+      const helpToggleButton = screen.getByText(messages.openHelpButtonLabel.defaultMessage);
+      fireEvent.click(helpToggleButton);
+      expect(screen.getByTestId('helpButtonRow')).toBeVisible();
+    });
+    it('should not show contect us button', () => {
```
Comment (connorhaugh) [inline src/footer/Footer.test.jsx:60]: Duplicate test names here

File: src/footer/Footer.test.jsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+      expect(screen.queryByTestId('accessibilityRequest')).toBeNull();
+    });
+    it('should show terms of service link', () => {
+      renderComponent('termsofserviceurl', null, null, false);
+      expect(screen.getByText('LMS')).toBeVisible();
+      expect(screen.queryByTestId('termsOfService')).toBeVisible();
+      expect(screen.queryByTestId('privacyPolicy')).toBeNull();
+      expect(screen.queryByTestId('accessibilityRequest')).toBeNull();
+    });
+    it('should show privacy policy link', () => {
```
Comment (connorhaugh) [inline src/footer/Footer.test.jsx:88]: duplicated tests here

**Review by connorhaugh — COMMENTED** [review]
Assumptions I have made for this review:
- The content/layout is correct (no mocks exist)
- You have made the choice to not put this in the footer library for a good reason.
- Code style is out of the window, as a lot of this is borrowed code.

File: src/footer/Footer.test.jsx (hunk 3 above)
Comment (KristinAoki) [reply src/footer/Footer.test.jsx:88]: Which ones are duplicate?

**Comment by KristinAoki** [comment]
> Assumptions I have made for this review:
> 
> * The content/layout is correct (no mocks exist)

Yes, content is matching the current footer in legacy studio. The coloring is set to match all the design patterns of the pages.

> * You have made the choice to not put this in the footer library for a good reason.

I am following the pattern of not including the header in the header library.

> * Code style is out of the window, as a lot of this is borrowed code.

@connorhaugh  what do you mean by code style?

File: src/footer/Footer.test.jsx (hunk 3 above)
Comment (connorhaugh) [reply src/footer/Footer.test.jsx:88]: Great question. ignore

File: src/footer/Footer.jsx [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        <Hyperlink destination={lmsBaseUrl}>LMS</Hyperlink>
+      </ActionRow>
+      <ActionRow className="mt-3 mx-4 pb-4 x-small">
+        {/*
+          Site operators: Please do not remove this paragraph! this attributes back to edX and
+            makes your acknowledgement of edX's trademarks clear.
+          Translators: 'edX' and 'Open edX' are trademarks of 'edX Inc.'. Please do not translate
+            any of these trademarks and company names.
+        */}
+        edX and Open edX, and the edX and Open edX logos are registered trademarks of <Hyperlink className="ml-1" destination="https://www.edx.org">edX Inc</Hyperlink>.
```
Comment (connorhaugh) [inline src/footer/Footer.jsx:113]: this should be a formatted message, as the note above suggests.

File: src/footer/Footer.jsx [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+        {/* need to add link to edx.org */}
```
Comment (connorhaugh) [inline src/footer/Footer.jsx:114]: you can delete this todo I think...

**Comment by connorhaugh** [comment]
> @connorhaugh what do you mean by code style?

cleaner code, principally breaking jsx into smaller components to be more readable. Not the end of the world in this case if it makes things harder.

**Review by connorhaugh — APPROVED** [review]

**Pull request merged**